collator/src/manager/blocks_cache.rs

```diff
@@ -149,33 +155,43 @@ impl BlocksCache {
                     mc_block_key,
                 )
             };
+
+            let processed_to = mc_block_entry
```

Comment:
Possibly we can use the same map type for `processed_to` across all the code without any perf cons? The conversion from `BTreeMap` to `HasMap` and vice versa for this data looks wired.